docs/manual/developer/04_style_guide.md

```diff
@@ -15,7 +15,36 @@
 * Should use the [merge commit method](https://docs.github.com/en/github/collaborating-with-pull-requests/incorporating-changes-from-a-pull-request/about-pull-request-merges)
 * Should use the GitHub Web UI to document and ensure code reviews are done correctly
 
-## All files
+
+## General Coding Style
+
+Prioritize the human-readability of code to its machine efficiency.
+In practice, this implies:
+
+* Keep functions and methods small, just a few of lines, so it is possible to understand what they do with minimal effort.
+  Prioritize the code readability over line count or code performance.
+* Don't mix low-level and high-level code in one function.
+  Extract functionality to separate functions or convert functions into classes with multiple methods to avoid this.
+* If a function contains try/except block, there should be no other logic present apart from exception handling.
+* Name variables and functions properly, don't hesitate to use longer identifiers.
+  Use explanatory variables to help the code to express itself -
+  assign a result of the operation into a variable that describes its meaning even if you intend to use it only once.
+* Don't copy-paste code, use e.g. jinja macros to reduce duplication.
+  Exception to this rule is code that is identical another piece of code only by coincidence, and there is a substantial probability that the code can diverge.
+* Include tests for your contribution.
+* Write comments if and only if there is no chance for the code to explain itself.
+* Don't put authorship information into the code, Git tracks authorship for you.
+  Use [SPDX IDs](https://spdx.dev/ids/) for license-related boilerplate in newly written code.
```

Comment:
Are we suggesting SPDX IDs for _all_ new (code) files or are we saying if a file needs it, we should add it? If the former, should we go back and apply it retroactively? A quick scan of some files in `ssg` and `build-scripts` says they're presently lacking, and IMO, I think the LICENSE in the root of the project applies fine enough to avoid SPDX-ing the project.

Reply:
Good point, the suggestion's intention was towards "don't copy-paste the whole license".